reana_job_controller/job_monitor.py

```diff
@@ -231,6 +234,21 @@ def get_job_logs(self, job_pod):
             logging.error("Unexpected error: {}".format(e))
             return None
 
+    def get_job_logs(self, job_pod) -> Optional[str]:
+        """Get job logs."""
+        logs = self._get_containers_logs(job_pod)
+
+        if job_pod.status.reason == "DeadlineExceeded":
+            if not logs:
+                logs = ""
+            logs += "\nJob was killed due to timeout."
```

Comment:
Cosmetics: Three remarks about the error message we sent back to users:

1. "Job was killed due to timeout" may be better said "Job was killed due to exceeding timeout". The idea is to be a bit more specific about "timeout" and that it is related to the job run time itself, not to some internal unspecified system timeout or something.

2. In case of out-of-memory situation, we expose the exact reason (`OOMKilled`). Perhaps we could expose here `DeadlineExceeded` as well? It could be handy for Kubernetes-savvy users.

3. If you have an easy way of getting timeout value, we could print it for further clarity?

Combining these three, it would give something like:

```
DeadlineExceeded

The job was killed due to exceeding timeout of 10 seconds.
```

Just jotting down cosmetic thoughts... We can simply do a minimal change and implement the first.

Reply:
1. This is more clear. Will update.

2.  `OOMKilled` is not exposed by REANA. k8s just adds this message at the end by default. But I can make something similar and just add one line with a reason. It will look like this:

```
...
Completed
DeadlineExceed

The job was killed due to exceeding the timeout for 10 seconds.
```

3. I think, I can get it from the job pod definition.

Reply:
> OOMKilled is not exposed by REANA.

What I mean above ist: `OOMKilled` string is textually exposed to users when they do `reana-client logs` command to inspect why a workflow failed.  So Kubernetes-savvy users can see the actual Kubernetes reason for the failure, instead of only seeing some REANA-formatted message.  

This is also how we expose `Completed` for successful jobs:

```console
$ reana-client logs -w roofit --filter step=gendata | tail -3

Completed
```

We could do the same for `DeadlineExceed` perhaps, so that we are consistent.